### PR TITLE
Only send bombardment combat when targets are visible

### PR DESCRIPTION
**Pull request by lmoureaux**
Bombarding cities was very powerful because it was showing every unit inside the city, something only diplomats and spies can normally do. Stop sending this around.

### CODE REVIEW

**Comment by lmoureaux** [comment]
Note that with this PR, there won't be any animation when bombarding.

**Comment by hugomflavio** [comment]
With this PR I don't see an animation when bombarding a city, but I still get a message for each unit bombarded inside the city, so that information is still leaking through?

image

**Comment by lmoureaux** [comment]
I didn't touch messages because they are handled in #2362

**Comment by hugomflavio** [comment]
but shouldn't these messages not be triggered if the server isn't sending the information to anyone but the defender? If the attacker receives this information, then the attacker can hack the client to make it visible.

**Comment by lmoureaux** [comment]
The messages are created and sent by the server and the client only displays them. I think your PR already removes them

**Comment by hugomflavio** [comment]
ooooh ok, my bad

File: server/unithand.cpp [hunk 1]
```diff
@@ -3592,7 +3606,8 @@ static void see_combat(struct unit *pattacker, struct unit *pdefender)
    Send combat info to players.
  */
 static void send_combat(struct unit *pattacker, struct unit *pdefender,
-                        int att_veteran, int def_veteran, int bombard)
```
Comment (hugomflavio) [inline server/unithand.cpp:3595]: You've removed `int bombard` but #2362 needs it

Comment (hugomflavio) [reply server/unithand.cpp:3595]: This seems to be the only compatibility issue between the two PRs - as far as I can tell

Comment (lmoureaux) [reply server/unithand.cpp:3595]: Oh, I see that #2362 uses this to not route bombardment towards the combat HUD. We may need to have a discussion about the client-side display of bombardment before we merge.

**Comment by hugomflavio** [comment]
Tested: Player C experience when player A bombs player B, both for a unit stack sitting outside and for units inside a city.
Confirmed that player C only sees one explosion - both for stacks and cities. No battle pop-up appeared.
client 3.1 with server 3.1
Note: Player A received 1 message per unit hit, but that's fixed by #2368

**Comment by hugomflavio** [comment]
Tested: New server with both old and new client.

When the new client bombards:
New client sees a single explosion, gets a message for each unit attacked. no popup.
Old client sees nothing at all.
Both for cities and unit stacks.

When the old client bombards:
New client sees a single explosion, gets no message/popup.
Old client does not see an explosion, but gets a message for each unit attacked. no popup.
Both for cities and unit stacks

**Comment by hugomflavio** [comment]
Just noticed one of the client (or both) were complaining about

```
[critical] freeciv21-client - Assertion nullptr != pplayer failed                                                  
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues  
```

I'll have to run them on separate consoles to see who's complaining.

Currently running the old server with the two clients and noticing a lot more of those than before.

**Comment by hugomflavio** [comment]
Testing: Old server with both new and old clients

When new client bombards:
Both clients receive battle pop-ups
Both clients see the unit sprites being bombard
Only attacker received messages
Same for cities and units.

When the old client bombards:
Both clients receive battle pop-ups
Both clients see the unit sprites being bombard
Only attacker received messages
Same for cities and units.

Again, some of these popped up every now and again, but didn't appear to affect the game
```
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
```

**Comment by hugomflavio** [comment]
On the PR client, destroying a city caused this:
```
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
[critical] freeciv21-client - Assertion nullptr != pplayer failed
[critical] freeciv21-client - Please report this message at https://github.com/longturn/freeciv21/issues
```

Bombarding is not related.

However, this only happened for the first city that was destroyed... Destroying other cities did not bring those up again.

The old client destroying a city of another nation didn't cause this either.

Old client didn't show assertion errors at all.

**Comment by hugomflavio** [comment]
nevermind, the master client can pop up the same kind of assertion errors too. Happened when destroying a city as well. It happened after I disconnected and reconnected the new client - maybe related?

**Pull request merged**